internal/plan/plan.go

```diff
@@ -40,12 +44,25 @@ type OptimisationValue struct {
 	Unit  int     `json:"unit"`
 }
 
-func (o OptimisationPlan) IsEmpty(logger *slog.Logger) bool {
+func (o OptimisationPlan) IsEmpty() bool {
```

Comment:
I like you removed the logger. Is a better signature.
This change, however, make me realise, I don't think we are running the linting rules on this project, are we? if the logger wasn't used before, the linter rule could have complained about it.

Reply:
Ugh, I've been faithfully running `golangci-lint run` locally before every commit, and we also have it in CI, but I don't have a settings file for it 🤦‍♀️

Reply:
I'll add that in a separate PR to avoid cluttering this one up with lint changes

Reply:
sure thing